composer.json

```
@@ -15,6 +15,7 @@
   "require": {
     "php": ">=7.2.0",
     "guzzlehttp/guzzle": "^6.0 || ^7.0",
+    "guzzlehttp/promises": "^2.0",
```

```suggestion
    "guzzlehttp/promises": "^1.4 || ^2.0",
```

What about supporting both? The static methods are there since 1.4: https://github.com/guzzle/promises/blob/2.0/CHANGELOG.md#140---2020-09-30

---

That sounds like a great idea. I'll have a look at it today.